dfc.getReplicas returned value changed in v6r20p12

The conversation started by private emails, but since I'm leaving, I add it here, such that @andresailer and @atsareg can keep looking at it.

**MAIL 1**
```
Hi Andrei,

your last PR regarding the VOPath in the DFC (https://github.com/DIRACGrid/DIRAC/pull/3814/files) has broken the interface:

If you run the integration tests (https://jenkins-dirac.web.cern.ch/view/DIRAC/job/Integration_SLC6/ws/testOutputs.txt/*view*/), you see that


======================================================================
FAIL: test_replicaOperations (__main__.ReplicaCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch/workspace/Integration_SLC6/ServerInstallDIR/DIRAC/tests/Integration/DataManagementSystem/Test_Client_DFC.py", line 541, in test_replicaOperations
    (testFile, result))
AssertionError: getReplicas failed, /vo.formation.idgrilles.fr/user/a/atsareg/testdir/testfile should be in Successful {'OK': True, 'Value': {'Successful': {'/vo.formation.idgrilles.fr/user/a/atsareg/testdir/testfile': {'otherSE': '', 'testSE': ''}}, 'Failed': {'/I/Dont/exist': 'No such file or directory'}, 'SEPrefixes': {'otherSE': '', 'FakeSE': '', 'testSE': '', 'nonExistingSe': ''}}}

----------------------------------------------------------------------
Ran 5 tests in 1.671s


The error message is misleading, but the actual error is that this is the returned dictionary  

'/vo.formation.idgrilles.fr/user/a/atsareg/testdir/testfile': {'otherSE': '', 'testSE': ''}

The otherSE and testSE values shouldn't be empty.

Without looking too much in details, I guess it comes from a missing 'else' statement here: https://github.com/DIRACGrid/DIRAC/pull/3814/files#diff-38faa43f37e89cea3917fb128720076dR126

I believe it breaks things for every VO that relies on the stored URL, rather than reconstructing
```

**Mail 2**

```
 Hi Chris,

  It looks like that in the integration test there are replicas in non-existent SE's. Before in this case
there were simply LFNs returned, I guess, which is also wrong. But the assertion was satisfied. So,
I think that the test itself should be improved to provide some definition for the otherSE and testSE.

  I do not think that the case where the URL is stored in the DFC is affected.
```

**Mail 3**

```
sorry I am not sure to understand.
"  It looks like that in the integration test there are replicas in non-existent SE's."
They do exist.
You can see above in the test

self.dfc.addFile({testFile: {'PFN': 'testfile',
                                          'SE': 'testSE',
                                          'Size': 123,
                                          'GUID': '1000',
                                          'Checksum': '0'}})
self.dfc.addReplica({testFile: {"PFN": "testFilePFN", "SE": "otherSE"}})

So that file is at existing SE.

Before, you are correct, what was returned was just the lfn, because the sePrefix was empty, and the code I pointed was storing "sePrefix + lfn".

So I agree, we should improve the test, and one of the SE should have a prefix.

But the other use case is valid as well, just imagine a S3 storage where the DFC will store the hash. There's no prefix there. So it should work as before. 
```

--- a/Resources/Catalog/FileCatalogClient.py
+++ b/Resources/Catalog/FileCatalogClient.py
@@ -121,10 +121,10 @@ class FileCatalogClient(FileCatalogClientBase):
           # The PFN was not returned, construct it on the fly
           # For some VO's the prefix can be non-standard
           voPrefix = seDict.get("VOPrefix", {}).get(se, {}).get(vo)
-          sePrefix = seDict.get(se)
+          sePrefix = seDict.get(se, '')
           prefix = voPrefix if voPrefix else sePrefix
-          if prefix:
-            lfnDict['Successful'][lfn][se] = prefix + lfn
+
+          lfnDict['Successful'][lfn][se] = prefix + lfn
 
     return S_OK(lfnDict)
 

